### PR TITLE
fix for #25 added version flag

### LINKED ISSUE #25: Add  a -V/--version option
It should just output the version and quit.

### PR DESCRIPTION
**Pull request by arthtyagi**
fixes #25

### CODE REVIEW

File: pyempaq/main.py [hunk 1]
```diff
@@ -157,6 +158,33 @@ def main():
         '-q', '--quiet',
         help="Only events of WARNING level and above will be tracked",
         action="store_const", dest="loglevel", const=logging.WARNING)
+    try:
+        # version = pkg_resources.get_distribution("pyempaq").
+        # version - this could be used to get the version of the "package"
+        # however, it's not necessarily the same as the version of the "module"
+        # parse the setup.py file to get the version
+        with open("setup.py", "rt") as fh:
```
Comment (facundobatista) [inline pyempaq/main.py:166]: Only try to get the version if `-V/--version` is used. Not always.

File: pyempaq/main.py [hunk 2]
```diff
@@ -157,6 +158,33 @@ def main():
...
+
+        # set the logging level
+        logging.basicConfig(level=args.loglevel)
+
+        # load the config
+        config = Config.from_file(args.source)
+
+        # pack
+        pack(config)
+    except Exception as err:
```
Comment (facundobatista) [inline pyempaq/main.py:184]: Avoid this big nonsense try/except block, please.

File: pyempaq/main.py [hunk 3]
```diff
@@ -157,6 +158,33 @@ def main():
...
+    try:
+        # version = pkg_resources.get_distribution("pyempaq").
+        # version - this could be used to get the version of the "package"
+        # however, it's not necessarily the same as the version of the "module"
+        # parse the setup.py file to get the version
+        with open("setup.py", "rt") as fh:
+            setup_py = fh.read()
+            version = re.search(r"version\s*=\s*['\"]([^'\"]+)['\"]", setup_py)[1]
+        from pyempaq.config_manager import Config
+        parser.add_argument(
```
Comment (facundobatista) [inline pyempaq/main.py:170]: When this is used a function should be run to get the version and show it.

File: pyempaq/main.py [hunk 4]
```diff
@@ -157,6 +158,33 @@ def main():
...
+            '-V', '--version',
+            action='version',
+            version=f"pyempaq {version}")
+        args = parser.parse_args()
```
Comment (facundobatista) [inline pyempaq/main.py:174]: From here to the end of the added code: Why all this is duplicated????

**Review by facundobatista — CHANGES_REQUESTED** [review]
Hello! I annotated some comments inline. Thanks!

File: pyempaq/main.py (hunk 2 above)
Comment (arthtyagi) [reply pyempaq/main.py:184]: Alright I'll avoid that "big nonsense" try/except block in the future lol. 

Will update this PR for now with the changes.

**Comment by arthtyagi** [comment]
I've made the essential changes and removed any unnecessary code blocks from the previous commits of this PR.

File: pyempaq/main.py [hunk 5]
```diff
@@ -21,6 +22,7 @@
 from pyempaq.config_manager import load_config, ConfigError
 
 
+
```
Comment (facundobatista) [inline pyempaq/main.py:25]: This change is not really needed. Please revert.

File: pyempaq/main.py [hunk 6]
```diff
@@ -157,6 +159,13 @@ def main():
         '-q', '--quiet',
         help="Only events of WARNING level and above will be tracked",
         action="store_const", dest="loglevel", const=logging.WARNING)
+    # if argument passed is -v, execute the function
+    parser.add_argument(
+            '-V', '--version',
+            action='version',
+            version=f"pyempaq {get_version()}")
```
Comment (facundobatista) [inline pyempaq/main.py:166]: As I said in the previous review, only try to get the version if -V/--version is used. Not always.

File: pyempaq/main.py [hunk 7]
```diff
@@ -157,6 +159,13 @@ def main():
...
+
+    # quit after displaying the version
```
Comment (facundobatista) [inline pyempaq/main.py:168]: This comment looks misplaced. That said, the functionality of exiting after showing version is not there.

File: pyempaq/main.py [hunk 8]
```diff
@@ -171,3 +180,10 @@ def main():
             logger.info(err, file=sys.stderr)
         exit(1)
     pack(config)
+
+def get_version():
+    """Get the version of the package."""
+    with open("setup.py", "rt") as fh:
+        setup_py = fh.read()
+        version = re.search(r"version\s*=\s*['\"]([^'\"]+)['\"]", setup_py)[1]
```
Comment (facundobatista) [inline pyempaq/main.py:188]: This line should be out of the context manager.

File: pyempaq/main.py [hunk 9]
```diff
@@ -171,3 +180,10 @@ def main():
...
+    return version
```
Comment (facundobatista) [inline pyempaq/main.py:189]: Missing a newline at the end of this line.

File: pyempaq/main.py [hunk 10]
```diff
@@ -157,6 +159,13 @@ def main():
         '-q', '--quiet',
         help="Only events of WARNING level and above will be tracked",
         action="store_const", dest="loglevel", const=logging.WARNING)
+    # if argument passed is -v, execute the function
+    parser.add_argument(
+            '-V', '--version',
```
Comment (facundobatista) [inline pyempaq/main.py:164]: These lines are over-indented.

**Review by facundobatista — CHANGES_REQUESTED** [review]
Thanks! I added some details in the review...

**Comment by cacrespo** [comment]
Hi @arthtyagi !
Do you need help to finish this issue ?
I'm trying to solve all milestone 0.3 issues and this is almost done.

**Comment by facundobatista** [comment]
This is done in #47.

**Pull request closed without merging**